app/javascript/hooks/mutations/sessions/useCreateSession.jsx

```diff
@@ -0,0 +1,26 @@
+import {useMutation, useQueryClient} from "react-query";
+import axios, { ENDPOINTS } from "../../../helpers/Axios";
+import {useNavigate} from "react-router";
+
+const createSession = (sessionUser) => axios.post(ENDPOINTS.signin, sessionUser)
+  .then((resp) => resp.data)
+  .catch((error) => console.log(error));
+
+export function useCreateSession(options){
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+
+  const mutation = useMutation(createSession, {
+    // Re-fetch the current_user and redirect to homepage if Mutation is successful.
+    onSuccess: () => {
+      queryClient.invalidateQueries('useSessions');
+      navigate('/rooms');
+    },
+    onError: (error) => {
+      console.log('mutate error', error);
+    },
+  });
+
+  const onSubmit = (user_data) => mutation.mutateAsync({ user: user_data }).catch(/*Prevents the promise exception from bubbling*/()=>{})
+  return { onSubmit, ...mutation }
```

Comment:
Why do we need to return onSubmit?
It does not seem to be used.

Why aren't we returning useMutation() straight up?

Reply:
I'm going to attempt to refactor this in the next PR

Reply:
I'm going to attempt to refactor this in the next PR